src/jit/lsra.cpp

```diff
@@ -5667,16 +5666,30 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
 
     if (availablePhysRegInterval != nullptr)
     {
+        RegRecord* physRegToUnassign = availablePhysRegInterval;
+#ifdef _TARGET_ARM_
+        if (!isUnassigned(availablePhysRegInterval, currentInterval))
```

Comment:
Given that this is used in a negation, I think you need to change the sense of this. That is, the method should be `isAssigned()` so that you don't have a confusing double negation.